src/fiber.cr

```diff
@@ -10,103 +44,77 @@ end
 class Fiber
   STACK_SIZE = 8 * 1024 * 1024
 
-  @@first_fiber : Fiber? = nil
-  @@last_fiber : Fiber? = nil
+  @@fibers = Thread::LinkedList(Fiber).new
```

Comment:
I imagined that a LinkedList module can be included and make the class a singleton linked list across ivars. And encapsulate the definition of next/previous there.

Reply:
You mean something like :
```crystal
module Thread::LinkedList(T)
  struct Root(T)
    def unsafe_each; end
    def push(node : T); end
    def delete(node : T); end
  end

  macro included
    class_getter all = Root(T).new
    property next : T?
    property previous : T?
  end
end

Reply:
Even more flattened:

```
module Thread::LinkedList(T)
  macro included
    @@first : T?
    @@last : T?
    @next : T?
    @previous : T?

    def self.unsafe_each; end
    def self.push(node : T); end
    def self.delete(node : T); end
  end
end
```

Reply:
Please don't jump to macros on this, this is an internal struct and the way it is in this PR is just fine.

Reply:
Ok